cmd/ebrelayer/contract/Peggy.sol

```diff
@@ -570,12 +576,16 @@ contract Peggy is Processor {
         whileActive()
         returns(bytes32 _id)
     {
-         //Actions based on token address type
+        string memory symbol;
+        
+        //Actions based on token address type
         if (msg.value != 0) {
           require(_token == address(0));
           require(msg.value == _amount);
+          symbol = "ETH";
```

Comment:
I don't understand this part. why would the symbol always be `ETH` if the value is not 0?

Reply:
msg.value is an ethereum-specific value that refers specifically to the amount of Ether sent in the transaction. Amount is a peggy-specific value that refers to the number the sender put in the function call, which should match either the etherum or token amount being sent.

It is possible for someone to send both ethereum and a token in the same transaction, but this check is for that if there is *any* ethereum sent, then we should only succeed if there is *only* ethereum sent, to avoid complexities. And in this scenario where there is only ethereum sent, ofc the symbol should be ETH.

@denali can you confirm?

Reply:
> It is possible for someone to send both ethereum and a token in the same transaction, but this check is for that if there is any ethereum sent, then we should only succeed if there is only ethereum sent, to avoid complexities.

shouldn't we replace the `else` for an `if` then?

Reply:
a comment about this on the code would be helpful as well

Reply:
@musnit Yes, that is correct. @fedekunze If I'm understanding your proposal correctly, two `if` then statements could be problematic as the second `if` statement would overwrite the ETH symbol declaration, so that the msg.value is effectively lost by the msg.sender. We could theoretically support both Ethereum and ERC20 tokens in the same transaction, but we would need still need token addresses and symbols for both (with all emitted in the LogLock event). There's a larger conversation to be had about batch asset locking, such that multiple ERC20 tokens can be locked in one transaction, but there are a myriad of considerations for batch asset transfers project wide which make it outside of the current scope.

That being said, I definitely agree that the contracts should be better commented for transparency.